pkg/hostagent/nodes.go

```diff
@@ -73,19 +73,28 @@ func (agent *HostAgent) initNodeInformerBase(listWatch *cache.ListWatch) {
 		AddFunc: func(obj interface{}) {
 			agent.nodeChanged(obj)
 		},
-		UpdateFunc: func(_ interface{}, obj interface{}) {
-			agent.nodeChanged(obj)
+		UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+			agent.nodeChanged(oldobj, newobj)
 		},
 		DeleteFunc: func(obj interface{}) {
 			agent.nodeDeleted(obj)
 		},
 	})
 }
 
-func (agent *HostAgent) nodeChanged(obj interface{}) {
+func (agent *HostAgent) nodeChanged(obj ...interface{}) {
 	updateServices := false
-
-	node := obj.(*v1.Node)
+	var node *v1.Node
+	if len(obj) == 2 {
+		oldnode := obj[0].(*v1.Node)
+		node = obj[1].(*v1.Node)
+		if !reflect.DeepEqual(node.ObjectMeta.Labels, oldnode.ObjectMeta.Labels) {
+			updateServices = true
+			agent.log.Info("Node label changed Update topology aware services")
```

Comment:
By setting this flag we are going to call updateAllServices(). I am guessing that should be good. If so, does this log need to change to just say - "Node label changed, updating services"?

Reply:
i will update the log